[TEP-0102] HTTPS Connection to Triggers ClusterInterceptor

savitaashture opened the pull request:
Issue: https://github.com/tektoncd/triggers/issues/871

As part of this TEP proposing to make default secure connection between ClusterInterceptor and EventListener.

savitaashture commented:
/assign @dibyom @khrm

savitaashture commented:
/kind tep

khrm commented on teps/0102-https-connection-to-triggers-interceptor.md:
```diff
@@ -0,0 +1,78 @@
+* No inputs required from user to run ClusterInterceptor server as `HTTPS` as everything is handled internally by Triggers.
+
+## Implementation Details
+At high level below are few implementation details
```
I think we also require changes in interceptor SDK so that custom interceptor can also be used in tls mode only.
So this work should start after https://github.com/tektoncd/triggers/pull/1312

savitaashture commented on teps/0102-https-connection-to-triggers-interceptor.md:
```diff
@@ -0,0 +1,82 @@
+* By default ClusterInterceptor run as `HTTPS`.
+* There won't be a support for `HTTP`.
+
+    Reason:
+    1. Support of `HTTPS` doesn't require any user intervention and providing secure connection is more preferred than insecure connection.
+    2. If we plan to support both `HTTP` and `HTTPS` we require input from user as well as need to add condition checks to execute based on user provided input. 
```
@dibyom added reason for not supporting `HTTP` connection

Please do let me know if require to add more info

dibyom commented on teps/0102-https-connection-to-triggers-interceptor.md:
```diff
@@ -0,0 +1,82 @@
+* Add new secret file to [config/interceptors](https://github.com/tektoncd/triggers/tree/main/config/interceptors) folder.
+* Update roles, clusterroles.
+* Changes to ClusterInterceptor server to run as `HTTPS`.
+* Changes to EventListener in order to connect with ClusterInterceptor securely.  
```
How will the EventListener find the cacert? In admission controllers, the WebhookConfiguration CR has (I think) a cacert field? Do we need to add something similar to the ClusterInterceptor CRD?

dibyom commented on teps/0102-https-connection-to-triggers-interceptor.md:
```diff
@@ -0,0 +1,82 @@
+* Changes to EventListener in order to connect with ClusterInterceptor securely.  
+
+## A look into the future
+* Providing a way to user to pass their own certificate to run ClusterInterceptor server.
```
If we add a `caCert` or something similar to the ClusterInterceptor CRD, this might be as simple as a user filling out this field. If it is missing, and the knative/pkg cert controller can generate the cert

dibyom reviewed (approved):
Overall looks the proposal looks good to me. +1 to doing the secure by default approach.

Just had one question on how to surface the `caCert` field easily.

savitaashture commented:
> Overall looks the proposal looks good to me. +1 to doing the secure by default approach.

@dibyom addressed review comments

> Just had one question on how to surface the `caCert` field easily.

I like the way we do it for `WebhookConfiguration` today 
I have included under `A look into the future`

dibyom commented on teps/0102-https-connection-to-triggers-interceptor.md:
```diff
@@ -0,0 +1,85 @@
+
+### User Stories
+
+* ClusterInterceptor calls are done using `HTTP` instead of `HTTPS` which is considered a security problem because
```
I'd say this makes more sense in the Motivation section than in the User Stories  section :)

dibyom commented:
> I like the way we do it for WebhookConfiguration today
I have included under A look into the future

Great, could we open a GH issue to track this feature?

I have one small comment on moving some text to the Motivation section instead. I think that can also be done in a new PR where we change the status to implementable given that we already have https://github.com/tektoncd/triggers/pull/1333 open

/lgtm
